apps/chat/templatetags/chat_tags.py

```diff
@@ -11,4 +11,10 @@
 def render_markdown(text):
     if not text:
         return ""
-    return linebreaksbr(mark_safe(markdown.markdown(text, extensions=[FencedCodeExtension()])))
+    text = markdown.markdown(text, extensions=[FencedCodeExtension()])
+    text = text.replace("</li>\n<li>", "</li><li>")
+    text = text.replace("<ol>\n<li>", "<ol><li>")
+    text = text.replace("</li>\n</ol>", "</li></ol>")
+    text = text.replace("<li>\n<p>", "<li><p>")
+    text = text.replace("</p>\n</li>", "</p></li>")
```

Comment:
What about using a regex to match all these cases? `<(.*?)>\\n<(.*?)>`

Reply:
Tried this. There is an exception to the `<(.*?)>\\n<(.*?)>` rule, which is two p tags. A newline between a closing and opening p tag looks very natural, so we don't want to match that. Will have to play around with regex a bit to see how we can do this. I'm not 100% sure if this is the only exception to the rule, but I think it is.

Reply:
@SmittieC could you add a comment with the outcome of our discussion for future reference. Specifically what elements the regex catches that we don't want it to.

Reply:
Sure! So the cases I added which should be excluded are the only ones (so far) to be addressed. There are cases that should not be changed, like \<p\>\\n\<\p\>. This means we cannot simply use regex to replace all instances where a newline character is present between html tags.